docs/source/dialogs.md

```diff
@@ -65,3 +69,22 @@ def process(self):
         ips = [entry[4][0] for entry in resolved if not '%' in entry[4][0]]
         self.produce(ResolvedHostname(name=hostname, ips=ips))
 ```
+
+### Explaining the dialogs processing mechanism during the upgrade
+
+The upgrade itself, from the operator's point of view, consists of 3 distinct stages: preupgrade, remediate and upgrade.
```

Comment:
```suggestion
The upgrade itself, from the operator's point of view, consists of 3 distinct stages: preupgrade, remediation and upgrade.
```

Reply:
I meant for preupgrade\upgrade to be verbs in this context, so remediate makes more sense imho